src/dissectors/ec_ospf.c

```diff
@@ -140,9 +152,24 @@ FUNC_DECODER(dissector_ospf)
         if (length > buflen)
                 return NULL;
 
-        DISSECT_MSG("OSPF-%s-%d:$netmd5$",
-                ip_addr_ntoa(&PACKET->L3.dst, tmp),
-                ntohs(PACKET->L4.dst));
+        if (auth_data_len == OSPF_AUTH_MD5_SIZE) {
+                DISSECT_MSG("OSPF-%s-%d:$netmd5$",
+                        ip_addr_ntoa(&PACKET->L3.dst, tmp),
+                        ntohs(PACKET->L4.dst));
+        } else if (auth_data_len == OSPF_AUTH_HMAC_SHA1_SIZE) {
+                type = 1;
+        } else if (auth_data_len == OSPF_AUTH_HMAC_SHA256_SIZE) {
+                type = 2;
+        } else if (auth_data_len == OSPF_AUTH_HMAC_SHA384_SIZE) {
+                type = 3;
+        } else if (auth_data_len == OSPF_AUTH_HMAC_SHA512_SIZE) {
+                type = 4;
+        }
+        if (type != 0) {
+                DISSECT_MSG("OSPF-%s-%d:$ospf$%d$",
+                        ip_addr_ntoa(&PACKET->L3.dst, tmp),
+                        ntohs(PACKET->L4.dst), type);
+        }
```

Comment:
Yes, john (JtR) uses this `type` value to automatically identify the hash type.

https://github.com/magnumripper/JohnTheRipper/blob/bleeding-jumbo/src/ospf_fmt_plug.c

Thanks! :)